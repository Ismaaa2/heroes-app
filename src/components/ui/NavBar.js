import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { Link, NavLink, useHistory } from "react-router-dom";
import { types } from "../../types/types";

export const Navbar = () => {
  
  const history = useHistory();

  const {
    user: { name },
    dispatch,
  } = useContext(AuthContext);

  const logout = () => {
    dispatch({ type: types.logout });


    history.replace("/login");
  };

  return (
    <nav className="navbar navbar-expand-sm p-3 navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">
        Heroes App
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/dc"
          >
            DC
          </NavLink>
          <NavLink
            exact
            activeClassName="active"
            className="nav-item nav-link"
            to="/search"
          >
            Search
          </NavLink>
        </div>
      </div>
      <div className="navbar-collapse flex-row-reverse collapse w-100 order-3 dual-collapse2">
        <ul
          style={{ alignItems: "center", justifyContent: "center" }}
          className="navbar-nav ml-auto"
        >
          <li style={{ marginRight: 20 }}>
            <p className="text-info" style={{ margin: 0, color: "royalblue" }}>{name}</p>
          </li>
          <li>
            <button
              className="nav-item nav-link btn btn-danger"
              onClick={logout}
            >
              Logout
            </button>
          </li>
        </ul>
      </div>
    </nav>
  );
};
