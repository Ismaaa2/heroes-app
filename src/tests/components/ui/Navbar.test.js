import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/NavBar";
import { types } from "../../../types/types";

describe("<Navbar />", () => {
  const mockHistory = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Ismael Bonilla",
    },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={mockHistory}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );
  test("deberÃ­a mostrar el nombre en la Navbar", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Ismael Bonilla");
  });

  test("deberia de ejecutarse el logout del NavBar", () => {
    wrapper.find("button").simulate("click");

    expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
    expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });

    expect(mockHistory.replace).toHaveBeenCalledWith("/login");
  });
});
