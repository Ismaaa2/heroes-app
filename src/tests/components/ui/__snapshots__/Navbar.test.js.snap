// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> deber√≠a mostrar el nombre en la Navbar 1`] = `
<nav
  className="navbar navbar-expand-sm p-3 navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Heroes App
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse flex-row-reverse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <li
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <p
          className="text-info"
          style={
            Object {
              "color": "royalblue",
              "margin": 0,
            }
          }
        >
          Ismael Bonilla
        </p>
      </li>
      <li>
        <button
          className="nav-item nav-link btn btn-danger"
          onClick={[Function]}
        >
          Logout
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
